plugins/languages/treesitter/treesitter.nix

```diff
@@ -411,10 +416,12 @@ helpers.neovim-plugin.mkNeovimPlugin config {
 
     extraFiles = mkIf cfg.nixvimInjections { "queries/nix/injections.scm".source = ./injections.scm; };
 
-    extraPlugins = mkIf (cfg.package != null) [
-      (mkIf cfg.nixGrammars (cfg.package.withPlugins (_: cfg.grammarPackages)))
-      (mkIf (!cfg.nixGrammars) cfg.package)
-    ];
+    extraPlugins =
+      lib.optionals (cfg.iconsPackage != null) [
+        cfg.iconsPackage
+      ]
+      ++ lib.optionals cfg.nixGrammars [ (cfg.package.withPlugins (_: cfg.grammarPackages)) ]
+      ++ lib.optionals (!cfg.nixGrammars) [ cfg.package ];
```

Comment:
It'd be nice if we still used mkIf/mkMerge/etc, but I think using `optional` and `if then else` ends up so much more readable that it's worth it?

Reply:
> I think we still need a `cfg.package != null` check somewhere.

I didn't think we had to since we don't accept null pluginPackage, still.

Reply:
> > > I think we still need a `cfg.package != null` check somewhere.
> > 
> > 
> > I didn't think we had to since we don't accept null pluginPackage, still.
> 
> That's precisely _why_ we need to check it here, to avoid accidentally adding a `[null]` definition.

It can't even evaluate that far if someone tried to provide a null cfg.package.

```bash
error: A definition for option `plugins.treesitter.package' is not of type `package'. Definition values:
       - In `<unknown-file>': null
```